static/app/components/replaysOnboarding/useCurrentProjectState.tsx

```diff
@@ -0,0 +1,92 @@
+import {useEffect, useMemo, useState} from 'react';
+import first from 'lodash/first';
+
+import {SidebarPanelKey} from 'sentry/components/sidebar/types';
+import {replayOnboardingPlatforms, replayPlatforms} from 'sentry/data/platformCategories';
+import PageFiltersStore from 'sentry/stores/pageFiltersStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {Project} from 'sentry/types';
+import useProjects from 'sentry/utils/useProjects';
+
+function useCurrentProjectState({currentPanel}: {currentPanel: '' | SidebarPanelKey}) {
+  const [currentProject, setCurrentProject] = useState<Project | undefined>(undefined);
+  const {projects, initiallyLoaded: projectsLoaded} = useProjects();
+  const {selection, isReady} = useLegacyStore(PageFiltersStore);
+
+  const isActive = currentPanel === SidebarPanelKey.ReplaysOnboarding;
+
+  // Projects where we have the onboarding instructions ready:
+  const projectsWithOnboarding = useMemo(
+    () =>
+      projects.filter(p => p.platform && replayOnboardingPlatforms.includes(p.platform)),
+    [projects]
+  );
+
+  // Projects that support replays, but we haven't created the onboarding instructions (yet):
+  const projectsWithoutFirstReplay = useMemo(
+    () =>
+      projects.filter(
+        p => p.platform && replayPlatforms.includes(p.platform) && p.hasReplays === false
+      ),
+    [projects]
+  );
+
+  useEffect(() => {
+    if (!isActive) {
+      setCurrentProject(undefined);
+    }
+  }, [isActive]);
+
+  useEffect(() => {
+    if (currentProject || !projectsLoaded || !projects.length || !isReady || !isActive) {
+      return;
+    }
+
+    if (!projectsWithoutFirstReplay) {
+      return;
+    }
+
+    if (selection.projects.length) {
+      const selectedProjectIds = selection.projects.map(String);
+      // If we selected something that has onboarding instructions, pick that first
+      const projectForOnboarding = projectsWithOnboarding.find(p =>
+        selectedProjectIds.includes(p.id)
+      );
+      if (projectForOnboarding) {
+        setCurrentProject(projectForOnboarding);
+      }
+
+      // If we selected something that supports replays pick that
+      const projectSupportsReplay = projectsWithoutFirstReplay.find(p =>
+        selectedProjectIds.includes(p.id)
+      );
+      if (projectSupportsReplay) {
+        setCurrentProject(projectSupportsReplay);
+      }
```

Comment:
Is it possible both of these conditions are true?

Reply:
yeah, it's possible. The idea is that we might have onboarding docs for only a subset of projects, so it's easier to tell if that's the case by doing a look up against the static list.

The list of onboarding docs should be a subset of the 'supported' list (which TS can help with).

both places should only include projects where `!project.hasReplays`, so we don't have to see onboarding docs if a project is already setup.